fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -40,6 +40,7 @@ def simulators(requested_os_type = "")
         end
 
         output.split(/\n/).each do |line|
+          break if line =~ /^-- Unavailable/
           next if line =~ /unavailable/
```

Comment:
This wasn't enough, as it only skipped to the next line, correct?

Reply:
`unavailable` (case insensitive) is meant to skip one simulator, not the entire group of simulators for some OS version. 
Because single simulator may be unavailable even os_version is available (e.g. see Xcode7 test spec) that line filtered such weird cases.

For Xcode 12 I get the output as below so even skipping one line (starting with`-- Unavailable`) would lead to an incorrect simulator `iPhone XR (8350A403-0E68-4F48-8ABA-C392F9CB8D7C) iOS 12.0` since in an `each` loop we initialize a simulator with a least recently observed valid iOS version.
```
-- iOS 12.0 --
    iPhone 5s (238C6D64-8720-4BFF-9DE9-FFBB9A1375D4) (Shutdown) 
    iPhone 6 (C68031AE-E525-4065-9DB6-0D4450326BDA) (Shutdown) 
-- Unavailable: com.apple.CoreSimulator.SimRuntime.iOS-12-1 --
    iPhone XR (8350A403-0E68-4F48-8ABA-C392F9CB8D7C) (Booted) 
    iPhone 5s (2721F3DC-A96B-4CA4-A9BD-B7CE3595BC82) (Shutdown) (unavailable, runtime profile not found) 
```



Optionally, I could just never initialize any simulator in a loop unless previously observed os_version was valid.